### ISSUE TITLE
Reduce Maven warnings during build

### ISSUE THREAD

**Issue by eeverman** [eeverman/andhow#444]
**_This is a good first task on the AndHow project if you know Maven well_**

Maven spits out a bunch of warnings during the build, making it hard to see when there really is a problem.  Most of this is around the `andhow` module.  The `andhow` is _special_ because it contains no code.  Instead, `anyhow` merges the jars in the `andhow-annotation-processor` and `andhow-core` jars.

`andhow-annotation-processor` is an annotation processor that inserts some files and classes into the target directory during compile.
`andhow-core` is a runtime library that looks for those generated files.

These two modules are logically separate, but its a bit dangerous to split them into separate dependencies because if you have one you really need the other.  To ensure that, they are merged into a single jar.
 
The build works as expected, but is _noisy_.  Are the Maven pom files setup correctly to merge these files?  Is there a better way?  Below is a snippet of the Maven output that this issue is intended to fix.

**_Note that during testing, AndHow is tested with invalid configuration which generates a lot of errors written to System.err and System.out.  This is intentional, though not ideal.  There is a separate task (#25) to address that._**

```
------------------------------------------------------------------------
Building andhow 0.4.1-SNAPSHOT
------------------------------------------------------------------------

--- maven-clean-plugin:2.5:clean (default-clean) @ andhow ---
Deleting /Users/ericevermanpersonal/workspaces/andhow/andhow/target

--- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven) @ andhow ---

--- maven-resources-plugin:2.6:resources (default-resources) @ andhow ---
Using 'UTF-8' encoding to copy filtered resources.
skip non existing resourceDirectory /Users/ericevermanpersonal/workspaces/andhow/andhow/src/main/resources

--- maven-compiler-plugin:3.8.0:compile (default-compile) @ andhow ---
No sources to compile

--- maven-resources-plugin:2.6:testResources (default-testResources) @ andhow ---
Using 'UTF-8' encoding to copy filtered resources.
skip non existing resourceDirectory /Users/ericevermanpersonal/workspaces/andhow/andhow/src/test/resources

--- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ andhow ---
No sources to compile

--- maven-surefire-plugin:2.12.4:test (default-test) @ andhow ---
No tests to run.

--- maven-jar-plugin:2.4:jar (default-jar) @ andhow ---
JAR will be empty - no content was marked for inclusion!
Building jar: /Users/ericevermanpersonal/workspaces/andhow/andhow/target/andhow-0.4.1-SNAPSHOT.jar

--- maven-shade-plugin:3.1.1:shade (default) @ andhow ---
Including org.yarnandtail:andhow-core:jar:0.4.1-SNAPSHOT in the shaded jar.
Could not get sources for org.yarnandtail:andhow-core:jar:0.4.1-SNAPSHOT:compile
Including org.yarnandtail:andhow-annotation-processor:jar:0.4.1-SNAPSHOT in the shaded jar.
Could not get sources for org.yarnandtail:andhow-annotation-processor:jar:0.4.1-SNAPSHOT:compile
Replacing original artifact with shaded artifact.
Replacing /Users/ericevermanpersonal/workspaces/andhow/andhow/target/andhow-0.4.1-SNAPSHOT.jar with /Users/ericevermanpersonal/workspaces/andhow/andhow/target/andhow-0.4.1-SNAPSHOT-shaded.jar
Replacing original source artifact with shaded source artifact.
Replacing /Users/ericevermanpersonal/workspaces/andhow/andhow/target/andhow-0.4.1-SNAPSHOT-sources.jar with /Users/ericevermanpersonal/workspaces/andhow/andhow/target/andhow-0.4.1-SNAPSHOT-shaded-sources.jar
Dependency-reduced POM written at: /Users/ericevermanpersonal/workspaces/andhow/andhow/dependency-reduced-pom.xml

--- maven-javadoc-plugin:3.0.1:jar (javadoc-jar) @ andhow ---
The POM for com.sun:tools:jar:sources:1.8 is missing, no dependency information available
The POM for com.sun:tools:jar:javadoc-resources:1.8 is missing, no dependency information available
no module descriptor for org.yarnandtail:andhow
no module descriptor for org.yarnandtail:andhow-core
no module descriptor for org.yarnandtail:andhow-annotation-processor
The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:javadoc' has not been previously called for the module: 'org.yarnandtail:andhow-core:jar:0.4.1-SNAPSHOT'. Trying to invoke it...
[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.
Creating fake javadoc directory to prevent repeated invocations: /Users/ericevermanpersonal/workspaces/andhow/andhow-core/target/apidocs
The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:javadoc' has not been previously called for the module: 'org.yarnandtail:andhow-annotation-processor:jar:0.4.1-SNAPSHOT'. Trying to invoke it...
[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.
Creating fake javadoc directory to prevent repeated invocations: /Users/ericevermanpersonal/workspaces/andhow/andhow-annotation-processor/target/apidocs
Error fetching link: /Users/ericevermanpersonal/workspaces/andhow/andhow-core/target/apidocs/package-list. Ignored it.
Error fetching link: /Users/ericevermanpersonal/workspaces/andhow/andhow-annotation-processor/target/apidocs/package-list. Ignored it.

100 warnings
Javadoc Warnings
/Users/ericevermanpersonal/workspaces/andhow/andhow-core/src/main/java/org/yarnandtail/andhow/AndHow.java:83: warning: no description for @return
* @return
^
... A whole bunch of javadoc warnings ...
```

**Comment by code4t2** [comment]
I'd like to take a shot at this issue too. I know my way around Maven fairly well, and I have decent Googling skills 😄

**Comment by code4t2** [comment]
Done. Please do look at #447 and tell me what you think.

**Comment by eeverman** [comment]
Fixed by #447
